prow/logrusutil/logrusutil.go

```diff
@@ -100,7 +103,15 @@ var censoredBytes = []byte(censored)
 // Censor replaces sensitive parts of the content with a placeholder.
 func (f CensoringFormatter) censor(content []byte) []byte {
 	for _, secret := range f.getSecrets().List() {
-		content = bytes.ReplaceAll(content, []byte(secret), censoredBytes)
+		if strings.HasSuffix(secret, "\n") || strings.HasPrefix(secret, "\n") {
```

Comment:
Please simplify this and use a logger instead of `fmt.Sprintf`:
```
log := logrus.NewEntry(logrus.StandardLogger())
for _, secret := range f.getSecrets().List() {
 if secret == ""{
    log.Warning("Secret is an empty string, ignoring")
    continue
 }
 if strings.HasSuffix(secret, "\n") || strings.HasPrefix(secret, "\n") {
  // TODO: Would it make more sense to use strings.TrimSpace?
  log.Warning("Secret has leading oder trailing linebreak, removing")
  secret = strings.Trim(secret, "\n")
 }
 content = bytes.ReplaceAll(content, []byte(secret), censoredBytes)
}
```

Also, if you do not want the next person to remove this straight away, please add dedicated tests for both these cases :)

Reply:
Done.